dev/WebView2/WebView2.cpp

```diff
@@ -0,0 +1,1942 @@
+    }
+    else if (deviceType == winrt::PointerDeviceType::Touch)
+    {
+        // Get pointer id for handling multi-touch capture
+        winrt::PointerPoint logicalPointerPoint{ args.GetCurrentPoint(*this) };
+        int32_t id = logicalPointerPoint.PointerId();
+        if (m_hasTouchCapture.count(id) > 0)
+        {
+            m_hasTouchCapture.erase(id);
+        }
+    }
+    else if (deviceType == winrt::PointerDeviceType::Pen)
+    {
+        m_hasPenCapture = false;
+    }
+}
+
+bool WebView2::ShouldNavigate(const winrt::Uri& uri)
+{
+    return uri != nullptr && uri.ToString() != m_stopNavigateOnUriChanged;
+}
+
+winrt::IAsyncAction WebView2::OnSourceChanged(winrt::Uri providedUri)
+{
+    // Trigger / wait for CWV2 creation if one isn't yet ready
+    if (!m_coreWebView)
+    {
+        // Creation request when one is already in flight (either explicit or implicit)
+        if (m_creationInProgressAsync)
+        {
+            MUX_ASSERT(m_isExplicitCreationInProgress || m_isImplicitCreationInProgress);
+
+            // If the in-flight creation is explicit (EnsureCWV2()), navigate to new source when that ends
+            // Otherwise it's implicit (another Source set), we are done; first Source will use updated Source value
+            if (m_isExplicitCreationInProgress)
+            {
+                // If EnsureCWV2() call is in flight, wait for it to complete then set Source
+                co_await *m_creationInProgressAsync;
+            }
+            else if (m_isImplicitCreationInProgress)
+            {
+                // If Set Source is in flight, nothing to do - it will apply latest source when it completes
+                co_return;
+            }
+        }
+        // First-time creation request
+        else
+        {
+            m_isImplicitCreationInProgress = true;
+            co_await CreateCoreObjects();
+        }
+    }
+
+    if (m_coreWebView)
+    {
+        // TODO_WebView2: Should we use RawUri() instead of AbsoluteUri()?
+        // Try to apply latest source (could have changed during the co_await's above)
+        if (m_isClosed)
+        {
+            throw winrt::hresult_error(RO_E_CLOSED, L"Cannot create CoreWebView2 (via Source) for Closed WebView2 element.");
+        }
+        const auto updatedUri = this->Source();
+        if (!updatedUri.Equals(providedUri) && ShouldNavigate(updatedUri))
+        {
+            m_coreWebView.Navigate(updatedUri.AbsoluteUri());
+        }
+        else
+        {
+            m_coreWebView.Navigate(providedUri.AbsoluteUri());
+        }
+    }
+
+    co_return;
+}
+
+HWND WebView2::GetHostHwnd() noexcept
+{
+    if (!m_xamlHostHwnd)
+    {
+        winrt::CoreWindow coreWindow = winrt::CoreWindow::GetForCurrentThread();
+        if (coreWindow)
+        {
+            auto coreWindowInterop = coreWindow.as<ICoreWindowInterop>();
+            winrt::check_hresult(coreWindowInterop->get_WindowHandle(&m_xamlHostHwnd));
+        }
```

Comment:
What happens in the context of OS Xaml islands / DWXS? I belive the CoreWindow will exist, but is not used for rendering... WebView2 will fail in some way. Should we early out to avoid unsafe behavior? Also, is there an issue/task to track WebView2 in island?

Reply:
This is just to get it working in UWP. We can have a follow-up to figure out how (if at all) to make it work in islands. I suspect there will be other things that need to be done.

Reply:
agree with getting it to work on UWP first and figuring out islands post "1.0".

Reply:
Having said that, an error message would be good if possible :)